slicebase/bin/slice-canary

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+YUMCONF="yum.conf"
+CANARYCONF="yum.canary.conf"
+YUMREPOD="yum.slice.d"
+CANARYREPOD="yum.canary.d"
+
+if [ -s /etc/slicename ]; then
+    SLICENAME=$(cat /etc/slicename)
+else
+    echo "Error: /etc/slicename is either empty or doesn't exist."
+    exit 1
+fi
+
+if [[ $EUID -ne 0 ]] ; then
+    echo "Error: please run slice-canary as root."
+    exit 1
+fi
+
+if ! service slicectrl stop &> /dev/null; then
+    echo "Error: slicectrl failed to stop all slice services."
+    exit 1
+fi
+
+# Function to clean up temporary yum configs
+function cleanup {
+    rm -rf /tmp/$CANARYCONF /tmp/$CANARYREPOD
+}
+
+# Copy the regular production yum configurations to new "canary" versions that
+# we can use to upgrade the slice without actually modifying the production
+# versions.
+cp /etc/$YUMCONF /tmp/$CANARYCONF
+cp -a /etc/$YUMREPOD /tmp/$CANARYREPOD
+
+# Modify the "canary" yum configurations to point to the M-Lab staging yum
+# repository.
+sed -i "s/\/etc\/${YUMREPOD//\./\\.}/\/tmp\/${CANARYREPOD//\./\\.}/" \
+    /tmp/$CANARYCONF
+sed -i '/slicebase\.centos/ { N;N; s/production-centos6/staging-centos6/ }' \
+    "/tmp/${CANARYREPOD}/slice.repo"
+
+# See if there is a newer version available before proceeding. It seems that
+# yum returns with code 100 when there is in fact an update available. We clean
+# all the cache files in advance to be sure that we're dealing with fresh
+# metadata.
+echo "* Checking to see if an update is available in the staging repository."
+yum -q clean all
+yum -c /tmp/yum.canary.conf check-update $SLICENAME > /dev/null
+if [ $? -ne 100 ]; then
+    echo "A newer version of slice package is not available."
+    echo "Is a newer version installed in the staging yum repository?"
+    cleanup
+    exit 1
+fi
+
+# Record the old version number of the slice package.
+OLDVER=$(yum list installed | grep $SLICENAME | awk '{print $2}')
```

Comment:
It's also possible to specify the package name:

    yum list installed $SLICENAME

The `-q` options limits some of the verbose output, but I don't know how to turn off the header from yum. Is it ok the OLDVER has the extra header value from column 2, "Packages"?

Also, does this command make any network calls? Is the version from yum preferable to the output from:

    rpm -q $SLICENAME